src/cpp/library/utility

```diff
@@ -56,7 +56,7 @@ pair<T1, T2> make_pair(T1 x, T2 y)
  */
 
 template <class T>
-typename std::remove_reference<T>::type &&move(T &&t) noexcept
+typename std::remove_reference<T>::type &&move(T &&t)
```

Comment:
Not specifically for this PR, I think we may need to decide whether or not to convert the function depending on the cpp version.

Reply:
I consider this:

````CPP
#if __cplusplus >= 201103L // Check if C++11 or later is being used
template <class T>
typename std::remove_reference<T>::type &&move(T &&t) noexcept
#else // Code for older C++ standards
template <class T>
typename std::remove_reference<T>::type &&move(T &&t)
#endif
{
  return static_cast<typename std::remove_reference<T>::type &&>(t);
}
````

Can we do better than this?

Reply:
Maybe we can identify and handle it in the frontend? because we have to deal with the standard library, may encounter various situations.

Reply:
The std::move utility didn't exist in C++98. 

The correct implementation should be:

````CPP
#if __cplusplus >= 201103L // Check if C++11 or later is being used
template <class T>
typename std::remove_reference<T>::type &&move(T &&t) noexcept
{
  return static_cast<typename std::remove_reference<T>::type &&>(t);
}
#endif

````

Reply:
> Maybe we can identify and handle it in the frontend? because we have to deal with the standard library, may encounter various situations. For this PR, I think it's good now.

I agree with you. 

Let's open an issue to remind us to handle these cases in the C++ frontend.